### DISCUSSION TITLE
BeeByte obfuscator on Android.

### DISCUSSION THREAD

**Discussion by Pequeno761** [community/community#123286] [Programming Help]
### Select Topic Area

How to  unobfuscate BeeByte Obfuscator

### Body

First, I don't know if I can insert discussions in my native language, so I did the translation (Portuguese-BR >>> English) using Google Translate :v
While analyzing the "Assembly-CSharp.dll" file using DNSpy, I came across a lot of obfuscated code, so I dug deeper and came across some classes "public class __BB_OBFUSCATOR_VERSION_3_9_2", "public static string ^`^]`^[]^` `]^`]`^[[`(byte[] ]]][`[][]]]^[]][]]`]`[], byte[] `^^[`]`]] []`[[^^`^]`[][)" obfuscated, "internal class ArrayCopy256" and "internal class ArrayCopy156" these classes appear 4 times throughout the .dll file and in specific security-related locations.
This means the files were obfuscated by Beebyte, correct? I looked for tools for deobfuscation but I couldn't use them. I completely unzipped the game and had access to the Unity project's Assets and Scripts and found more obfuscated scripts...
Can you guys help me with this?
The "Assembly-CSharp.dll" will be available for download https://drive.google.com/file/d/1oJ-fjCMYfXRCtar9zRNrM34-XgVVVMAt/view?usp=sharing, I used DNSpy to dig through it.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Pequeno761 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
